src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -1117,6 +1117,7 @@ PhiNode* PhiNode::split_out_instance(const TypePtr* at, PhaseIterGVN *igvn) cons
     }
   }
   Compile *C = igvn->C;
+  ResourceMark rm;
```

Comment:
I would suggest leaving this change to a separate (possible starter) RFE.

Reply:
The problem with leaving this to a separate RFE is that the idealization changes in this changeset trigger a memory consumption bug related to this missing `ResourceMark`. Specifically, without the `ResourceMark` we now hit the 1GB memory limit in quite a number of tests. Therefore, it seems counterproductive to not fix as part of this changeset?

Reply:
Fair enough, I was not aware of the interaction. Please leave this conversation unresolved as additional insight for future reviewers.